### PR TITLE
Gst iet jsonchanges

### PR DESCRIPTION
**Pull request by manoj-freyr**

### CODE REVIEW

File: iet.so/include/action.h [hunk 1]
```diff
@@ -75,7 +75,7 @@ class iet_action: public rvs::actionbase {
 
  protected:
     //! TRUE if JSON output is required
-    bool bjson;
+    bool bjson = false;
```
Comment (jkottiku) [inline iet.so/include/action.h:78]: Looks good

File: iet.so/include/iet_worker.h [hunk 2]
```diff
@@ -288,6 +288,7 @@ class IETWorker : public rvs::ThreadBase {
     bool iet_tp_flag;
     //mtex
     std::mutex mtx_blas_done;
+    bool endtest = false;
```
Comment (jkottiku) [inline iet.so/include/iet_worker.h:291]: Looks good

File: gst.so/src/gst_worker.cpp [hunk 3]
```diff
@@ -598,12 +609,8 @@ uint64_t GSTWorker::time_diff(
 void GSTWorker::log_to_json(const std::string &key, const std::string &value,
```
Comment (jkottiku) [inline gst.so/src/gst_worker.cpp:597]: Can we move this function to rvs util or separate json util file ?

File: gst.so/src/gst_worker.cpp [hunk 4]
```diff
@@ -64,6 +65,19 @@ using std::string;
 
 bool GSTWorker::bjson = false;
 
+// common util
+void *json_node_create(std::string module_name, std::string action_name,
```
Comment (jkottiku) [inline gst.so/src/gst_worker.cpp]: Can we move this function to rvs util or separate json util file ?

File: iet.so/src/iet_worker.cpp [hunk 5]
```diff
@@ -69,7 +69,7 @@
 using std::string;
 
 bool IETWorker::bjson = false;
-bool endtest = false;
+//bool endtest = false;
```
[the remark this answers is not in the document]
Comment (manoj-freyr) [reply iet.so/src/iet_worker.cpp:72]: Done

File: gst.so/include/gst_worker.h [hunk 6]
```diff
@@ -33,6 +33,9 @@
 #define GST_RESULT_PASS_MESSAGE         "true"
 #define GST_RESULT_FAIL_MESSAGE         "false"
 
+void *json_node_create(std::string module_name, std::string action_name,
```
[the remark this answers is not in the document]
Comment (manoj-freyr) [reply gst.so/include/gst_worker.h]: in the updated files, it is already in rvs_util

File: gst.so/src/action.cpp [hunk 7]
```diff
@@ -536,6 +537,32 @@ int gst_action::get_all_selected_gpus(void) {
 
     return 0;
 }
+/**
+ * @brief flushes target and dtype fields to json file
+ * @return 
+ */
+
+void gst_action::json_add_primary_fields(){
```
[the remark this answers is not in the document]
Comment (manoj-freyr) [reply gst.so/src/action.cpp:545]: the primary fields usually vary between modules and hence i have put it here. Once other modules are refactored we ca take larger subset of necessary ones and move them to util

File: gst.so/src/gst_worker.cpp (hunk 3 above)
Comment (manoj-freyr) [reply gst.so/src/gst_worker.cpp:597]: the primary fields usually vary between modules and hence i have put it here. Once other modules are refactored we ca take larger subset of necessary ones and move them to util

File: gst.so/src/gst_worker.cpp (hunk 4 above)
Comment (manoj-freyr) [reply gst.so/src/gst_worker.cpp]: in the updated files, it is already in rvs_util

**Review by jkottiku — APPROVED** [review]

**Review by jkottiku — APPROVED** [review]

**Pull request merged**